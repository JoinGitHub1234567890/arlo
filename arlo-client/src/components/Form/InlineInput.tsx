import React from 'react'
import styled from 'styled-components'

const InputContainer = styled.div`
    width: 50%;
    display: flex; 
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 10px;
`

const InputLabel = styled.label`
    display: inline-block;
`

interface Props {
    label?: String,
    value?: any,
    onChange?: any,
    name?: any,
    defaultValue?: any
}

const InlineInput = (props: Props) => {
    const {label, value, onChange, ...restProps} = props
    return (
        <InputContainer>
            <InputLabel>{label}</InputLabel>
            <input value={value} onChange={onChange} {...restProps}/>
        </InputContainer>
    )
}

export default InlineInput