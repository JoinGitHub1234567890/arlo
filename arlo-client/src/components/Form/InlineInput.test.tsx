import React from 'react'
import { render } from '@testing-library/react'
import InlineInput from './InlineInput'

it('renders corretly', () => {
  const { container } = render(<InlineInput />)
  expect(container).toMatchSnapshot()
})
