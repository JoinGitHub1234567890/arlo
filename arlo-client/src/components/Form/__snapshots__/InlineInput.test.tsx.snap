// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders corretly 1`] = `
<div>
  <div
    class="sc-bdVaJa fCyZCP"
  >
    <label
      class="sc-bwzfXH kNDdYo"
    />
    <input
      value=""
    />
  </div>
</div>
`;
